pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/HelixBrokerStarter.java

```diff
@@ -95,7 +95,7 @@ public HelixBrokerStarter(String brokerHost, String helixClusterName, String zkS
     _pinotHelixProperties = DefaultHelixBrokerConfig.getDefaultBrokerConf(pinotHelixProperties);
 
     if (brokerHost == null) {
-      brokerHost = NetUtil.getHostAddress();
+      brokerHost = NetUtil.getHostnameOrAddress();
```

Comment:
whats the difference between getHostAddress and getHostnameOrAddress. Is this backward compatible? Should we first check for getHostAddress and if it returns null or empty get HostnameOrAddress. 
Another option is to add a startup config param

Reply:
getHostAddress will return ip.
getHostnameOrAddress will try to return hostname first and fallback to ip if failed.
This is definitely not backward-compatible, I think I will make it config based, though ideally people should have their own wrapper startable to set corresponding configs.